compose/material3/material3/src/jsWasmMain/kotlin/androidx/compose/material3/PlatformDateFormat.jsWasm.kt

```diff
@@ -211,6 +212,19 @@ internal actual object PlatformDateFormat {
 
         return js("new Intl.Locale(locale).weekInfo.firstDay").unsafeCast<Int>()
     }
+
+    actual fun is24HourFormat(locale: CalendarLocale): Boolean {
+
+        if (!isIntlSupported)
+            return false
+
+        @Suppress("UNUSED_VARIABLE")
+        val localeTag = locale.toLanguageTag()
+
+        //hourCycles is a list containing 'h24' or 'h23'
+        return js("new Intl.Locale(localeTag).hourCycles.join().indexOf('h2') >= 0")
```

Comment:
Firefox seems to not support `hoursCycles`. It doesn't support the method either - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getHourCycles#supported_hour_cycle_types

It supports only `hourCycle` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/hourCycle

But `hourCycle` is defined only when it's explcitly was declared in Locale constructor. At the same time `hoursCycles` returns an array `['h12']` for `Intl.Locale("en-US")`.

So we need to add more conditioncs. Maybe use `hoursCycles` when it's possible. If not, then `hourCycle`. And if it's undefined, then return false (I guess we have no other option).

As I understand, the current code would crash on firefox. Did you try to run the test on firefox?

Reply:
Yep, it would crash, as well as `weekInfo`. Added more conditions and exception checks

Reply:
Added fallback method

Reply:
Cleaned up a little bit, removed `isIntlSupported` check as we catch any exception anyway